apps/src/lab2/levelEditors/validations/EditValidations.tsx

```diff
@@ -108,6 +108,20 @@ const EditValidations: React.FunctionComponent<EditValidationsProps> = ({
     setValidations([...validations, createNewValidation()]);
   };
 
+  const duplicateValidation = (validationToDuplicate: Validation) => {
```

Comment:
Should we wrap in `useCallback`?

Reply:
I believe `useCallback` isn't needed here because `duplicateValidation` is used internally, and not passed to children, so identity changes don’t matter.
By contrast, the new panel functions in the other PR are passed to child components so memoizing them helps prevent avoidable updates. (VSC actually insisted I do it for that one but not here.)

Reply:
Hmm yeah, I mean strictly speaking I guess this is passed to an inner component (`EditValidations`) right? But if it is memoized, its dependencies would probably be `validations` which implies that it'll get recreated every time validations change anyway?

Reply:
Yup, exactly.